tests/test_pyghooks.py

```diff
@@ -337,3 +340,37 @@ def mock_os_listxattr(*args, **kwards):
         color_token, color_key = color_file(file_path, os.lstat(file_path))
 
         assert color_key == "ca"
+
+
+@pytest.mark.parametrize(
+    "name, styles, refrules",
+    [
+        ("test1", {}, {}),  # empty styles
+        (
+            "test2",
+            {Token.Literal.String.Single: "#ff0000"},
+            {Token.Literal.String.Single: "#ff0000"},
+        ),  # Token
+        (
+            "test3",
+            {"Token.Literal.String.Single": "#ff0000"},
+            {Token.Literal.String.Single: "#ff0000"},
+        ),  # str key
+        (
+            "test4",
+            {"Literal.String.Single": "#ff0000"},
+            {Token.Literal.String.Single: "#ff0000"},
+        ),  # short str key
+    ],
+)
+def test_register_custom_style(name, styles, refrules):
```

Comment:
Should this be moved to `test_tools.py`?

Reply:
Well the `register_custom_style` in tools calls this + the ANSI counterpart, so the `pygments` related part is in `pyghooks` and the ANSI related is in `ansi_colors`.
This might be renamed to `register_custom_pygments_style` here (in `pyghooks` and `test_pyghooks`).
I want to add tests for `tools.register_custom_style` as well (but I probably won't have time to do it today).

Reply:
Yeah that seems like the right way to go.  

> I probably won't have time to do it today

That's OK! Thanks so much for putting this in!

Reply:
Unit tests added for `xonsh.tools.register_custom_style`.